src/lib/components/Collapsible.svelte

```diff
@@ -0,0 +1,186 @@
+<script lang="ts">
+  import { afterUpdate, createEventDispatcher } from "svelte";
+  import IconExpandMore from "$lib/icons/IconExpandMore.svelte";
+  import { i18n } from "$lib/stores/i18n";
+  import { replacePlaceholders } from "$lib/utils/i18n.utils";
+
+  export let id: string | undefined = undefined;
+  export let initiallyExpanded = false;
+  export let maxContentHeight: number | undefined = undefined;
+
+  export let iconSize: "small" | "medium" = "small";
+  export let expandButton = true;
+  export let externalToggle = false;
+  export let wrapHeight = false;
+
+  // Minimum height when some part of the text-content is visible (empirical value)
+  const CONTENT_MIN_HEIGHT = 40;
+  const dispatch = createEventDispatcher();
+
+  export let expanded = initiallyExpanded;
+  let offsetHeight: number | undefined;
+  let userUpdated = false;
+  let maxHeight: number | undefined;
+
+  const dispatchUpdate = () => dispatch("nnsToggle", { expanded });
+
+  export const toggleContent = () => {
+    userUpdated = true;
+    expanded = !expanded;
+    dispatchUpdate();
+  };
+
+  const calculateMaxContentHeight = (): number => {
+    if (maxContentHeight !== undefined) return maxContentHeight;
+    const height = offsetHeight === undefined ? 0 : offsetHeight;
+    return height < CONTENT_MIN_HEIGHT ? CONTENT_MIN_HEIGHT : height;
+  };
+  const maxHeightStyle = (height: number | undefined): string =>
+    height === undefined ? "" : `max-height: ${height}px;`;
+  // In case of `initiallyExpanded=true` we should avoid calculating `max-height` from the content-height
+  // because the content in the slot can be initialized w/ some delay.
+  const updateMaxHeight = () => {
+    if (userUpdated) {
+      maxHeight = expanded ? calculateMaxContentHeight() : 0;
+    } else {
+      maxHeight = initiallyExpanded ? maxContentHeight : 0;
+    }
+  };
+  // Avoid to show scroll if not necessary
+  const overflyYStyle = (height: number | undefined): string =>
+    height === undefined || maxContentHeight === undefined
+      ? "overflow-y: hidden;"
+      : height < maxContentHeight
+      ? "overflow-y: hidden;"
+      : "overflow-y: auto;";
+
+  // recalculate max-height after DOM update
+  afterUpdate(updateMaxHeight);
+  let toggleView: string;
+  $: toggleView = expanded ? "collpase" : "expand";
+</script>
+
+<div
+  data-tid="collapsible-header"
+  id={id !== undefined ? `heading${id}` : undefined}
+  role="term"
+  class={`header ${externalToggle ? "external" : ""}`}
+  on:click={() => (externalToggle ? undefined : toggleContent())}
+>
+  <div class="header-content">
+    <slot name="header" />
+  </div>
+  {#if expandButton}
+    <button
+      class="collapsible-expand-icon"
+      class:size-medium={iconSize === "medium"}
+      class:expanded
+      data-tid="collapsible-expand-button"
+      aria-expanded={expanded}
+      aria-controls={id}
+      title={replacePlaceholders($i18n.core.toggle, {
```

Comment:
I would just remove `replacePlaceholders` and use words "Collapse" and "Expand".

```
title={expanded ? i18n.core.collapse : i18.core.expand}
```

Doing so we can spare copying the function `replacePlaceholders`